### PR TITLE
Cumulus 817 documentation

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-817](https://bugs.earthdata.nasa.gov/browse/CUMULUS-817)

## Changes

* Adds documentation for features introduced in https://github.com/nasa/cumulus/pull/656

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing
- [x] Update CHANGELOG

### CODE REVIEW

**Comment by Jkovarik** [comment]
Ignore all non-documentation changes on review if present, as they're introduced in #656

File: docs/features/lambda_dead_letter_queue.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+# Cumulus Dead Letter Queues
+
+## startSF SQS queue
+
+The [workflow-trigger](../workflows/workflow-triggers) for the startSF queue has a [Redrive Policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sqs-queues-redrivepolicy.html) set up that directs any failed attempts to pull from the workflow start queue to a SQS queue [Dead Letter Queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html).
+
+This queue can then be monitored for failures to initiate a workflow.   Please note that workflow failures will not show up in this queue, only repeated failure to trigger a workflow.
+
+## Named Lambda Dead Letter Queues
```
Comment (laurenfrederick) [inline docs/features/lambda_dead_letter_queue.md]: Newline after this

File: packages/deployment/app/config.yml [hunk 2]
```diff
@@ -13,6 +13,13 @@ default:
   message_adapter_repo: cumulus-message-adapter
   message_adapter_filename: cumulus-message-adapter.zip
 
+  customCompilation: true
+  injectWorkflowLambdaAliases: true
+
+  DLQDefaultTimeout: 60
+  DLQDefaultMessageRetentionPeriod: 1209600
```
Comment (laurenfrederick) [inline packages/deployment/app/config.yml]: Just like in the other PR review, comment that this is 14 days or whatever.

File: packages/deployment/lib/kes.js [hunk 3]
```diff
@@ -287,6 +295,27 @@ class UpdatedKes extends Kes {
...
+   */
+  addLambdaDeadLetterQueues() {
+    const lambdas = this.config.lambdas;
+    Object.keys(lambdas).forEach((key) => {
+      const lambda = lambdas[key];
+      if (lambda.namedLambdaDeadLetterQueue) {
+        console.log(`Adding named dead letter queue for ${lambda.name}`);
+        const queueName = `${lambda.name}DeadLetterQueue`;
+        this.config.sqs[queueName] = {
+          MessageRetantionPeriod: this.config.DLQDefaultMessageRetentionPeriod,
```
Comment (laurenfrederick) [inline packages/deployment/lib/kes.js]: I was thinking, should we have a default here so they don't have to set it if they don't want to?

File: packages/deployment/app/config.yml (hunk 2 above)
Comment (Jkovarik) [reply packages/deployment/app/config.yml]: @laurenfrederick  sorry, I should probably have made this a clean branch: 

https://github.com/nasa/cumulus/pull/662#issuecomment-434445610

File: docs/features/lambda_dead_letter_queue.md (hunk 1 above)
Comment (Jkovarik) [reply docs/features/lambda_dead_letter_queue.md]: 👍

File: packages/deployment/lib/kes.js (hunk 3 above)
Comment (Jkovarik) [reply packages/deployment/lib/kes.js]: @laurenfrederick  I'm not sure what you mean on this comment -- as this is a the doc PR, I'll see about chatting offline about this and update the ticket requirements if needed for close.

**Pull request merged**